ConsentViewController/Classes/SourcePointClient/GDPRPrivacyManagerViewResponse.swift

```diff
@@ -13,32 +13,61 @@ struct GDPRVendor: Decodable {
     }
 
     enum Keys: String, CodingKey {
-        case policyUrl, vendorId, name
+        case policyUrl, legIntUrl, vendorId, name
         case iabId
-        case iabSpecialPurposes, iabFeatures, iabSpecialFeatures
+        case iabSpecialPurposes, iabSpecialPurposesObjs, iabFeatures, iabSpecialFeatures, iabDataCategories
         case description, cookieHeader
         case vendorType
         case consentCategories, legIntCategories, disclosureOnlyCategories
     }
 
     struct Category: Decodable {
-        let type: GDPRCategory.CategoryType
+        let type: GDPRCategory.CategoryType?
         let iabId: Int?
         let name: String
+        var retention: String?
+        
+        enum CodingKeys: CodingKey {
+            case type
+            case iabId
+            case name
+            case retention
+        }
+        
+        init(from decoder: Decoder) throws {
+            let container: KeyedDecodingContainer<GDPRVendor.Category.CodingKeys> = try decoder.container(keyedBy: GDPRVendor.Category.CodingKeys.self)
+            self.type = try container.decodeIfPresent(GDPRCategory.CategoryType.self, forKey: GDPRVendor.Category.CodingKeys.type)
+            self.iabId = try container.decodeIfPresent(Int.self, forKey: GDPRVendor.Category.CodingKeys.iabId)
+            self.name = try container.decode(String.self, forKey: GDPRVendor.Category.CodingKeys.name)
+            do {
+                self.retention = try container.decodeIfPresent(String.self, forKey: GDPRVendor.Category.CodingKeys.retention)
+            } catch DecodingError.typeMismatch {
+                self.retention = try String(container.decode(Int.self, forKey: GDPRVendor.Category.CodingKeys.retention))
+            }
+        }
```

Comment:
@Nevazhnovu we can write a fancy extension for the code above (and unit tests for it). Something like:
```swift
extension KeyedDecodingContainer {
    public func decodeStringOrIntIfPresent(
        forKey key: KeyedDecodingContainer<K>.Key
    ) throws -> String? {
        do {
            return try decodeIfPresent(String.self, forKey: key)
        } catch DecodingError.typeMismatch {
            return try String(decode(Int.self, forKey: key))
        }
    }
}
```
And use:
```swift
self.retention = try container.decodeStringOrIntIfPresent(forKey: .retention)
```

Reply:
Sounds good! Even less code to maintain :)